sound/soc/sof/loader.c

```diff
@@ -145,14 +146,12 @@ int snd_sof_parse_module_memcpy(struct snd_sof_dev *sdev,
 			return -EINVAL;
 		}
 		snd_sof_dsp_block_write(sdev, sdev->mmio_bar, offset,
-					(void *)((u8 *)block + sizeof(*block)),
-					block->size);
+					block + 1, block->size);
 
 		/* minus body size of block */
 		remaining -= block->size;
 		/* next block */
-		block = (struct snd_sof_blk_hdr *)((u8 *)block + sizeof(*block)
-			+ block->size);
+		block = (struct snd_sof_blk_hdr *)(block + 1 + block->size);
```

Comment:
not sure this one is correct? the addition of sizeof looks very suspicious here. Let's leave the code as it was before for now.

Reply:
@plbossart fixed now.